examples/pigweed-app/nrfconnect/README.md

```diff
@@ -0,0 +1,208 @@
+third_party/pigweed, to see what is needed for integrating Pigweed to CHIP, as
+well as a precursor to CHIP functionalities like on-device testing.
+
+The example makes use of the CMake build system to generate the ninja build
+script. The build system takes care of invoking the CHIP library build with all
+necessary flags exported from the Zephyr environment.
+
+<a name="building"></a>
+
+## Building
+
+### Using Docker container
+
+> **Important**:
+>
+> Due to
+> [certain limitations of Docker for MacOS](https://docs.docker.com/docker-for-mac/faqs/#can-i-pass-through-a-usb-device-to-a-container)
+> it is impossible to use the Docker container to communicate with a USB device
+> such as nRF 52840 DK. Therefore, MacOS users are advised to follow the
+> [Using Native shell](#using-native-shell) instruction.
+
+The easiest way to get started with the example is to use nRF Connect SDK Docker
+image for CHIP applications. Run the following commands to start a Docker
+container:
+
+        $ mkdir ~/nrfconnect
+        $ mkdir ~/connectedhomeip
+        $ docker pull nordicsemi/nrfconnect-chip
+        $ docker run --rm -it -v ~/nrfconnect:/var/ncs -v ~/connectedhomeip:/var/chip -v /dev/bus/usb:/dev/bus/usb --device-cgroup-rule "c 189:* rmw" nordicsemi/nrfconnect-chip
+
+> **Note**:
+>
+> -   `~/nrfconnect` can be replaced with an absolute path to nRF Connect SDK
+>     source directory in case you have it already installed.
+> -   Likewise, `~/connectedhomeip` can be replaced with an absolute path to
+>     CHIP source directory.
+> -   `-v /dev/bus/usb:/dev/bus/usb --device-cgroup-rule 'c 189:* rmw`
+>     parameters can be omitted if you're not planning to flash the example onto
+>     hardware. The parameters give the container access to USB devices
+>     connected to your computer such as the nRF52840 DK.
+> -   `--rm` flag can be omitted if you don't want the container to be
+>     auto-removed when you exit the container shell session.
+
+If you use the container for the first time and you don't have nRF Connect SDK
+and CHIP sources downloaded yet, run `setup` command in the container to pull
+the sources into directories mounted as `/var/ncs` and `/var/chip`,
+respectively:
+
+        $ setup --ncs 83764f
+        /var/ncs repository is empty. Do you wish to check out nRF Connect SDK sources [83764f]? [Y/N] y
+        ...
+        /var/chip repository is empty. Do you wish to check out Project CHIP sources [master]? [Y/N] y
+        ...
+
+Now you may build the example by running the commands below in the Docker
+container:
+
+        $ cd /var/chip/examples/lock-app/nrfconnect
+        $ west build -b nrf52840dk_nrf52840
+
+If the build succeeds, the binary will be available under
+`/var/chip/examples/lock-app/nrfconnect/build/zephyr/zephyr.hex`. Note that
+other operations described in this document like flashing or debugging can also
+be done in the container.
+
+### Using native shell
```

Comment:
This version works better, can we document the checkout in "Supported nRF Connect SDK versions" better?

i.e, provide the steps

```
mkdir ~/nrfconnect
cd ~/nrfconnect
west init -m https://github.com/nrfconnect/sdk-nrf --mr 83764f01f04f58c80bb1ae1d3e0a640496f79b49
west update
```

How do you change the `--mr` hash used by `west update` ?

Reply:
I described it, as you requested. Actually this is a duplication of information located under the link attached in the instruction, but anyway I hope that now it will be easier for someone to set it up.